light/client.go

```diff
@@ -1006,23 +1077,39 @@ func (c *Client) compareNewHeaderWithWitnesses(h *types.SignedHeader) error {
 }
 
 func (c *Client) compareNewHeaderWithWitness(errc chan error, h *types.SignedHeader,
-	witness provider.Provider, witnessIndex int) {
+	witness provider.Provider, witnessIndex int, now time.Time) {
 
 	altH, err := witness.SignedHeader(h.Height)
 	if err != nil {
 		errc <- errBadWitness{err, noResponse, witnessIndex}
 		return
 	}
-
 	if err = altH.ValidateBasic(c.chainID); err != nil {
 		errc <- errBadWitness{err, invalidHeader, witnessIndex}
 		return
 	}
 
+	altVals, err := witness.ValidatorSet(h.Height)
```

Comment:
do you just want to use `c.signedHeaderAndValSetFromWitness` instead for these 4 functions